content/numerical/FastFourierTransformMod.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Author: chilli
+ * Date: 2019-04-25
+ * License: CC0
+ * Source: http://neerc.ifmo.ru/trains/toulouse/2017/fft2.pdf
+ * Description: Can be used for convolutions modulo arbitrary integers.
+ * Is essentially a higher precision FFT.
+ * Is safe in practice as long as $(|a|+|b|)\sqrt{\max(a)\max(b)} < \mathtt{\sim} 10^{15}$.
```

Comment:
`$(|a|+|b|)M < \mathtt{\sim} 10^{15}$`, right? You can shorten the description a bit into "Higher precision FFT, can be used for convolutions modulo arbitrary integers. Safe in practice ..."

Reply:
maybe we should change "in practice" to "for random numbers"

Reply:
I think the claim is a bit stronger than that. I think that statement is a pretty common CP belief.

Proof by consensus, if you will.

Reply:
I think we're good for this issue?